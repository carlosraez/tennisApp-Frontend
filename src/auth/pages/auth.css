
   .containerLogin {
      text-align: center;
   }
   label {
      margin-bottom: 10px;
      margin-top: 10px;
      font-weight: 600;
      width: 100%;
      text-align: initial;
   }
   .button-Sign {
      margin-top: 30px;
   }
   .loginGoogle {
      width: 100%;
      cursor: pointer;
      padding: 5px;
    }
  
    .chooseLogin {
      font-size: 20px;
      font-weight: 600;
      margin-top: 10px;
    }
   
    .linkGoToResgister {
     margin-top: 10px;
    }
   
    span {
     font-size: 12px;
     display: none;
     text-align: left;
     padding-top: 5px;
    }
   
    input:invalid[focused="true"] {
     border: 1px solid red 
    }

    select:invalid[focused="true"] {
      border: 1px solid red 
     }

     select:invalid[focused="true"] ~ span{
      color: red;
      font-weight: 600;
      display: block;
   }
   
    input:invalid[focused="true"] ~ span{
       color: red;
       font-weight: 600;
       display: block;
    }
 

   .containerLogin {
      display: flex;
      justify-content: center;
      align-items: center;
      text-align: center;
   }
 
   .containerLogin {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100vh;
    }

    @media (min-width: 576px) { 
      .containerLogin {
         height: 100vh;
      }
     }
 

 
 

 

 

 