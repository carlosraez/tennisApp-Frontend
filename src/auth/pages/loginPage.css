 .containerLogin {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    text-align: center;
 }

 label {
    margin-bottom: 10px;
    margin-top: 10px;
    font-weight: 600;
    width: 100%;
    text-align: initial;
 }

 .button-Sign {
    margin-top: 30px;
 }

 .loginGoogle {
   width: 100%;
   cursor: pointer;
   padding: 5px;
 }

 span {
   margin-left: 10px;
 }

 .chooseLogin {
   font-size: 20px;
   font-weight: 600;
   margin-top: 10px;
 }

 .linkGoToResgister {
  margin-top: 10px;
 }