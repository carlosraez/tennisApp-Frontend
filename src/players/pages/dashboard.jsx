import React from 'react'

import registerImage from '../../assets/registerPlayer.jpg';
import { CardDashboard } from '../components/cardDashboard';

export const Dashboard = () => {

 

  return (
    <div className="container">
      <div className="row">
        <div className="col-md-6">
          <CardDashboard 
            linkHref={'/player-register'} 
            registerImage={registerImage}
            title={'Register Player'}
            description={'Register a new player'}
            textLink={'Register'}
            />
        </div>
        <div className="col-md-6">
           <CardDashboard 
           linkHref={'/players'}
           registerImage={registerImage}
           title={'Register Player'}
            description={'Register a new player'}
            textLink={'Register'}
           />
        </div>
      </div>
    </div>
  )
}
