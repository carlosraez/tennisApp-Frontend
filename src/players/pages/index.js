export * from './dashboard';
export * from './players';