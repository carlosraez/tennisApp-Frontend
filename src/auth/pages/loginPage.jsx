import React, { useState, useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux/es/exports';

import './auth.css';
import logoRaquet from '../../assets/raquetsLogo.jpg';
import { en } from '../../i18n/index';
import { checkingLoginAuthentication } from '../../store/auth/thunk';
import { checkingCredentials } from '../../store/auth/authSlice';
import { InputsForm } from '../components/inputsForm';

export const LoginPage = () => {
  const [formValid, setFormValid] = useState(false);
  const [formValues, setFormValues] = useState({
    email:'',
    password:'',
  });
  const [focused, setfocused] = useState(false)

  const dispatch = useDispatch();
  const inputRef = useRef();
  
  const { errorMessage } = useSelector(state => state.auth);

  const { email, password } = formValues;
  
   /**
   * @description - This function is used to handle the input change
   * @returns {void} 
   */
  const handleInputChange = ({ target }) => { 
    setFormValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  /**
   * @description - This function is used put the focus on the input when the page is loaded and load errors messages
   * @returns {void}
   */
  const handleFocus = () => { 
     setfocused(true)
  }

   useEffect(() => {
   formValidations()
   timeCheckingMessage()
   }, [email, password,]);

   /**
    * @description - This function is used to time the message of the error
    * received from the server.
    * @returns {void}
    */
   const timeCheckingMessage = () => {
      if (errorMessage) {
       const msg = ''
        setTimeout(() => {
          dispatch(checkingCredentials(msg));
        }, 3000);
       }
   }
  
  /**
   * data inputsFormLogin 
   * @type {Array}
   */
  const inputsFormLogin = [
    {
      label: en.inputLabelUser,
      type: 'email',
      required: true,
      placeholder: en.inputPlaceHolderMail,
      ariaLabel: 'userEmail',
      name: 'email',
      value: email,
      errorMessage: en.errorInputEmail,
      required: true,
    },
    {
      label: en.passwordCreate,
      type: 'password',
      placeholder: en.inputPlaceHolderPasswrod,
      ariaLabel: 'userPassword',
      value: password,
      required: true,
      errorMessage: en.errorInputPassword,
      pattern: '^[0-9]{3,6}/*$',
      name: 'password',
    },
  ];
 
 /**
  * FormValid - checks if the form is valid
  * @returns {void}
  */
 const formValidations = () => {
  const form = inputRef.current
  const inputsInvalid = form.querySelectorAll('input:invalid');
  if (inputsInvalid.length <= 0 && errorMessage === null) {
       setFormValid(true);
  }
 }
  
 /**
  * Dispacht Login authentication
  * @param {e} e event to prevent default
  * @returns {void} dispatch checkingLoginAuthentication
  */
 const getFormSubmit = e => {
  e.preventDefault();
  if (formValid) {
    dispatch(checkingLoginAuthentication(formValues));
  }
};

   /**
   * InputsForm Login
   * @returns {JSX.Element}
   */
  const getFormInputs = () => {
    return (
    <InputsForm 
     inputsFormLogin={inputsFormLogin}  
     handleInputChange={handleInputChange}
     handleFocus={handleFocus}
     focused={focused}
    />)
  };

   /**
   * Button submit
   * @returns {JSX.Element}
   */
  const getLoginButton = () => {
    return (
      <button disabled={!formValid} type="submit" className="w-80 btn btn-lg btn-primary button-Sign">
        {en.titleLogin}
      </button>
    );
  };

  
  /**
   * Link to register page
   * @returns {JSX.Element}
   */
  const getLinkToLogin = () => {
    return (
      <p className="linkGoToResgister">
        <Link color="inherit" to="/auth/register">
          {en.notHaveAccount}
        </Link>
      </p>
    );
  };

  /**
   * Error message if user is not registered or has wrong credentials
   * @returns {JSX.Element}
   */
  const getError = (msg) => {
    return (
      <div>
        <div className="alert alert-danger" role="alert">
           {msg}
        </div>
      </div>
    );
  };

  /**
   * Logo of the application
   * @returns {JSX.Element}
   */
  const getLogo = () => {
    return (
      <img
        className="mb-4"
        src={logoRaquet}
        width={200}
        height={200}
        alt="logo"
      />
    );
  };
  
  return (
    <div className="w-100 m-auto containerLogin">
      <form ref={inputRef} onSubmit={getFormSubmit}>
        <div className='container'>
          <div className='row'> 
          <div className='col-xs-100 col-md-6'>
           {errorMessage && getError(errorMessage)}
           {getFormInputs()}
           {getLoginButton()}
          </div>
          <div className='col-xs-100 col-md-6'>
            <h1>{en.titleRegister}</h1>
            {getLogo()}
            <p className="chooseLogin">{en.chooseLogin}</p>
            {getLinkToLogin()}
          </div>
        </div>
        </div>
      </form>
    </div>
  );
};