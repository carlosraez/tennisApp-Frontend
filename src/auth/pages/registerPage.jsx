import React, { useState, useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux/es/exports';

import './auth.css';
import logoRaquet from '../../assets/raquetsLogo.jpg';
import { en } from '../../i18n/index';
import { checkingRegisterAuthentication } from '../../store/auth/thunk';
import { checkingCredentials } from '../../store/auth/authSlice';
import { InputsForm } from '../../components/inputsForm';

export const RegisterPage = () => {
  const [formValid, setFormValid] = useState(false);
  const [formValues, setFormValues] = useState({
    name:'',
    email:'',
    password:'',
    confirmPassword:''
  });
  const [focused, setfocused] = useState(false)

  const dispatch = useDispatch();
  const inputRef = useRef();
  
  const { errorMessage } = useSelector(state => state.auth);

  const {name, email, password, confirmPassword } = formValues;

  /**
   * @description - This function is used to handle the input change
   * @returns {void} 
   */
  const handleInputChange = ({ target }) => { 
    setFormValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  /**
   * @description - This function is used put the focus on the input when the page is loaded and load errors messages
   * @returns {void}
   */
  const handleFocus = () => { 
     setfocused(true)
  }

   useEffect(() => {
   formValidations()
   timeCheckingMessage()
   }, [email, name, password, confirmPassword]);

    /**
    * @description - This function is used to time the message of the error
    * received from the server.
    * @returns {void}
    */
   const timeCheckingMessage = () => {
    if (errorMessage) {
      console.log('me ejcuto');
      const msg = ''
      console.log(msg);
      setTimeout(() => {
        dispatch(checkingCredentials(msg));
      }, 3000);
     }
   }

  /**
  * data inputsFormRegister 
  * @type {Array}
  */
  const inputsFormRegister = [
    {
      label: en.inputLabelUserName,
      type: 'text',
      placeholder: en.inputPlaceHolderUserName,
      ariaLabel: 'userName',
      name: 'name',
      value: name,
      errorMessage: en.errorInputName,
      pattern: '^[A-z]{3,16}/*$',
      required: true,
    },
    {
      label: en.inputLabelUser,
      type: 'email',
      placeholder: en.inputPlaceHolderMail,
      ariaLabel: 'userEmail',
      name: 'email',
      value: email,
      errorMessage: en.errorInputEmail,
      required: true,
    },
    {
      label: en.passwordCreate,
      type: 'password',
      placeholder: en.inputPlaceHolderPasswrod,
      ariaLabel: 'userPassword',
      value: password,
      errorMessage: en.errorInputPassword,
      pattern: '^[0-9]{6,6}/*$',
      name: 'password',
      required: true,
    },
    {
      label: en.repitePassword,
      type: 'password',
      placeholder: en.inputPlaceHolderPasswrod,
      ariaLabel: 'confirmPassword',
      value: confirmPassword,
      errorMessage: en.errorInputConfirmPassword,
      pattern: password,
      name: 'confirmPassword',
      required: true,
    },
  ];

 /**
  * FormValid - checks if the form is valid
  * @returns {void}
  */
 const formValidations = () => {
  const form = inputRef.current
  const inputsInvalid = form.querySelectorAll('input:invalid');
  if (inputsInvalid.length <= 0 && errorMessage === null || errorMessage === '') {
       setFormValid(true);
  }
 }
  
  /**
  * Dispacht Register authentication
  * @param {e} e event to prevent default
  * @returns {void} dispatch checkingLoginAuthentication
  */
  const getFormSubmit = e => {
    e.preventDefault();
    if (formValid) {
     dispatch(checkingRegisterAuthentication(formValues));
     }
  };

  /**
   * InputsForm Login
   * @returns {JSX.Element}
   */
  const getFormInputs = () => {
    return (
      <InputsForm
       inputsForm={inputsFormRegister}  
       handleInputChange={handleInputChange}
       handleFocus={handleFocus}
       focused={focused}
      />)
  };

  /**
  * Button submit
  * @returns {JSX.Element}
  */
  const getLoginButton = () => {
    return (
      <button disabled={!formValid} type="submit" className="w-80 btn btn-lg btn-primary button-Sign">
        {en.registerButton}
      </button>
    );
  };

  /**
  * Link to register page
  * @returns {JSX.Element}
  */
  const getLinkToLogin = () => {
    return (
      <p className="linkGoToResgister">
        <Link color="inherit" to="/auth/login">
          {en.haveAccount}
        </Link>
      </p>
    );
  };

  /**
  * Error message if user is not registered or has wrong credentials
  * @returns {JSX.Element}
  */
  const getError = (msg) => {
    return (
      <div>
        <div className="alert alert-danger" role="alert">
           {msg}
        </div>
      </div>
    );
  };

  /**
  * Logo of the application
  * @returns {JSX.Element}
  */
  const getLogo = () => {
    return (
      <img
        className="mb-4"
        src={logoRaquet}
        width={200}
        height={200}
        alt="logo"
      />
    );
  };
  
  return (
    <div className="w-100 m-auto containerLogin">
      <form ref={inputRef} onSubmit={getFormSubmit}>
        <div className='container'>
          <div className='row'> 
          <div className='col-xs-100 col-md-6'>
           {errorMessage && getError(errorMessage)}
           {getFormInputs() }
           {getLoginButton()}
          </div>
          <div className='col-xs-100 col-md-6'>
            <h1>{en.titleRegister}</h1>
            {getLogo()}
            <p className="chooseLogin">{en.chooseLogin}</p>
            {getLinkToLogin()}
          </div>
        </div>
        </div>
      </form>
    </div>
  );
};


