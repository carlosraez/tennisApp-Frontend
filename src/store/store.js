import { configureStore } from '@reduxjs/toolkit';
import { authSlice } from './auth/';
import { playerSlice } from './player/';

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    player: playerSlice.reducer,
  },
});
