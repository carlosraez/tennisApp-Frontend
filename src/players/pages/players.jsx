import React, { useState, useRef, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux/es/exports';
import { Link } from 'react-router-dom';

import { InputsForm } from '../../components/inputsForm';
import { ListPlayers } from '../../components/listPlayers';
import { en } from '../../i18n';
import { addPlayer, deletePlayerList } from '../../store/player';

export const Players = () => {
  const [formValid, setFormValid] = useState(false);
  const [formValues, setFormValues] = useState({
    name:'',
    tennisShot:'',
    location:'',
    birthday:'',
    level:'',
    });
  const [focused, setfocused] = useState(false)

  useEffect(() => {
    formValidations()
  }, [formValues]);
  
  const players  = useSelector(state => state.player.players);
  const dispatch = useDispatch();
  const inputRef = useRef();

  const {name, tennisShot, location, birthday, level } = formValues;

  const inputsFormPlayer = [
    {
      label: en.inputLabelUserName,
      type: 'text',
      placeholder: en.playerName,
      ariaLabel: 'userName',
      name: 'name',
      value: name,
      errorMessage: en.errorInputName,
      pattern: '^[A-z]{3,16}/*$',
      required: true,
    },
    {
      label: en.bestTennisShot,
      ariaLabel: 'userTennisShot',
      name: 'tennisShot',
      value: tennisShot,
      options: [ en.driveShot, en.serveShot, en.reverse,],
      errorMessage: en.errorTennisShotInput,
      isSelect: true,
      required: true,
    },
    {
      label: en.locationPlayer,
      type: 'text',
      placeholder: en.inputPlaceHolderLocation,
      ariaLabel: 'locationPlayer',
      name: 'location',
      value: location,
      errorMessage: en.errorLocationInput,
      pattern: '^[A-z]{3,16}/*$',
      required: true,
    },
    {
      label: en.birthday,
      type: 'date',
      ariaLabel: 'BirthDayPlayer',
      name: 'birthday',
      value: birthday,
      errorMessage: en.errorBirthDayInput,
      pattern: '/^(0?[1-9]|[12][0-9]|3[01])[\/\-](0?[1-9]|1[012])[\/\-]\d{4}$/',
      required: true,
    },
    {
      label: en.selectLevel,
      isSelect: true,
      ariaLabel: 'level',
      name: 'level',
      value: level,
      errorMessage: en.errorSelectInput,
      options:  [ en.begginer,en.intermediate ,en.advanced, en.tournament ],
      required: true,
    }
  ]

  const hanldeFormPlayer = (e) => {
    e.preventDefault();
    if (formValid) {
      dispatch(addPlayer(formValues));
      setFormValues({
        name:'',
        tennisShot:'',
        location:'',
        birthday:'',
        level:'',
      });
    }
  }

  const formValidations = () => { 
    const form = inputRef.current
    const inputsInvalid = form.querySelectorAll('input:invalid');
    inputsInvalid.length === 0 ? setFormValid(true) : setFormValid(false);
  }

  /**
   * 
   * @param {object} e event to prevent default  
   */
  const handleInputChange = (e) => {
     const { name, value } = e.target;
      setFormValues({ ...formValues, [name]: value });
  }

   /**
   * @description - This function is used put the focus on the input when the page is loaded and load errors messages
   * @returns {void}
   */
    const handleFocus = () => { 
      setfocused(true)
   }


   /**
   * Button save form player
   * @returns {JSX.Element}
   */
  const getSavePlayerButton = () => {
    return (
      <button type="submit" className="btn btn-primary mt-3">
        {en.titleSavePlayer}
      </button>
    );
  };

  /**
   * @description - This function is used to show all the form inputs to register a new player
   * @returns {JSX.Element}
   */
  const getFormRegisterPlayer = () => { 
     return (
       <InputsForm 
         inputsForm={inputsFormPlayer}  
         handleInputChange={handleInputChange}
         handleFocus={handleFocus}
         focused={focused}
        />
     )
  } 

  /**
   * @describe - This function is used to delete a player from the list
   * @param {any} e event to prevent default
   */
  const handleDeletePlayer = (e) => {
    const id = e.target.id;
    const tennisPlayerDelete = players[id] 
    dispatch(deletePlayerList(tennisPlayerDelete));
  }

  
  
  /**
   * @describe - Table list off tenis in data base
   * @returns {JSX.Element}
   */
  const getListPlayers = () => {
      return (
        <table className="table">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Shot</th>
            <th scope="col">Location</th>
            <th scope="col">Level</th>
            <th scope="col">Update</th>
            <th scope="col">Delete</th>
          </tr>
        </thead>
         <tbody>
            {players.map((player, index) => {
              return (
                <ListPlayers 
                index={index}
                player={player}
                handleDelete={handleDeletePlayer}
                />
              )
            })}
         </tbody>
      </table>
      )
  }

  return (
      <div className="container">
        <div className="row">
          <Link color="inherit" to="/dashboard">{en.backToDashboard}</Link>
          <h1>{en.titleRegisterPlayer}</h1>
          <div className="col-xs-100 col-md-6">
            <form ref={inputRef} onSubmit={hanldeFormPlayer}>
              {getFormRegisterPlayer()}
              {getSavePlayerButton()}
            </form>
          </div>
          <div className="col-xs-100 col-md-6">
              {getListPlayers()}
          </div>
        </div>
     </div>
  )
};
