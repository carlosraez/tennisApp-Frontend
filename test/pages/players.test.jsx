
import { render as rtlRender } from "@testing-library/react"
import { Provider } from "react-redux"
import  { MemoryRouter } from "react-router-dom"

import { Dashboard } from "../../src/players/pages/dashboard"
import { store } from "../../src/store/store"

const render = component => 
    rtlRender(
        <Provider store={store}>
            <MemoryRouter>
                {component}
            </MemoryRouter>
      </Provider>
    )


describe('test att players dashboard', () => { 
    test('should be show the dashboard equals', () => {
        const { container } = render(<Dashboard  />)
        expect(container).toMatchSnapshot()
    })   
})