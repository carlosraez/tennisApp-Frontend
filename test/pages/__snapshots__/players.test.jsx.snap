// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test att players dashboard should be show the dashboard 1`] = `
<div>
  <div
    aria-label="dashboardApp"
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="..."
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Register Player
            </h5>
            <p
              class="card-text"
            >
              Register a new player
            </p>
            <a
              class="btn btn-primary"
              href="/player-register"
            >
              Register
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="..."
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Register Player
            </h5>
            <p
              class="card-text"
            >
              Register a new player
            </p>
            <a
              class="btn btn-primary"
              href="/players"
            >
              Register
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test att players dashboard should be show the dashboard equals 1`] = `
<div>
  <div
    aria-label="dashboardApp"
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="..."
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Register Player
            </h5>
            <p
              class="card-text"
            >
              Register a new player
            </p>
            <a
              class="btn btn-primary"
              href="/player-register"
            >
              Register
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="..."
            class="card-img-top"
            src=""
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Register Player
            </h5>
            <p
              class="card-text"
            >
              Register a new player
            </p>
            <a
              class="btn btn-primary"
              href="/players"
            >
              Register
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
